File: Python/0026RemoveDuplicatesFromSortedArray.py
def removeDuplicates(nums : list[int]) -> int:
    low = 0
    high = 1
    uniqueCount = 1

    while(high < len(nums)):
        if nums[high] == nums[high - 1]:
            high += 1
            continue

        nums[low + 1] = nums[high]

        low += 1
        high += 1
        uniqueCount += 1
    
    return uniqueCount

nums = [1, 1, 1, 2, 2, 3]
print(removeDuplicates(nums))
print(nums)

File: C/0026RemoveDuplicatesFromSortedArray.c
#include <stdio.h>

int removeDuplicates(int* nums, int numsSize){
    int low = 0;
    int high = 1;
    int uniqueCount = 1;

    while(high < numsSize){
        if(nums[high] == nums[high - 1]){
            high++;
            continue;
        }

        nums[low + 1] = nums[high];

        low++;
        high++;
        uniqueCount++;
    }

    return uniqueCount++;
}

int main(){

    int nums[] = {1, 1, 1, 2, 2, 3};
    int numsSize = 6;

    printf("%d", removeDuplicates(nums, numsSize));

    return 0;
}